lib/libtsd/tsd_task.c

```diff
@@ -326,10 +326,14 @@ tsd_task_start(struct tsd_task *t)
 			if (ret != 0)
 				_exit(1);
 		}
-		if (getgid() != getegid())
-			(void)setgid(getgid());
-		if (getuid() != geteuid())
-			(void)setuid(getuid());
+		if (getgid() != getegid() && setgid(getgid()) != 0) {
+			ERROR("failed to reset process group");
+			goto fail;
```

Comment:
You are now in the child process. The `fail` target is in the parent code. Using `goto fail` here will result in the child process “becoming” a new, duplicate tsdfx master process which will compete with its parent (and possibly occupy a task slot in the parent forever). Assuming that `setgid()` or `setuid()` will either never fail or always fail, this change turns tsdfx into a fork bomb that also heavily exercises the file system.

The correct behavior is to just fall through to the `_exit(1)` at the bottom of the child block.